ibc-core/ics04-channel/src/handler/timeout.rs

```diff
@@ -215,37 +215,46 @@ where
-                return Err(PacketError::InvalidPacketSequence {
-                    given_sequence: msg.packet.seq_on_a,
-                    next_sequence: msg.next_seq_recv_on_b,
+        let next_seq_recv_verification_result = match chan_end_on_a.ordering {
+            Order::Ordered => {
+                if msg.packet.seq_on_a < msg.next_seq_recv_on_b {
+                    return Err(PacketError::InvalidPacketSequence {
+                        given_sequence: msg.packet.seq_on_a,
+                        next_sequence: msg.next_seq_recv_on_b,
+                    }
+                    .into());
                 }
-                .into());
+                let seq_recv_path_on_b =
+                    SeqRecvPath::new(&msg.packet.port_id_on_b, &msg.packet.chan_id_on_b);
+
+                client_state_of_b_on_a.verify_membership(
+                    conn_end_on_a.counterparty().prefix(),
+                    &msg.proof_unreceived_on_b,
+                    consensus_state_of_b_on_a.root(),
+                    Path::SeqRecv(seq_recv_path_on_b),
+                    msg.packet.seq_on_a.to_vec(),
+                )
+            }
+            Order::Unordered => {
+                let receipt_path_on_b = ReceiptPath::new(
+                    &msg.packet.port_id_on_b,
+                    &msg.packet.chan_id_on_b,
+                    msg.packet.seq_on_a,
+                );
+
+                client_state_of_b_on_a.verify_non_membership(
+                    conn_end_on_a.counterparty().prefix(),
+                    &msg.proof_unreceived_on_b,
+                    consensus_state_of_b_on_a.root(),
+                    Path::Receipt(receipt_path_on_b),
+                )
+            }
+            Order::None => {
+                return Err(ContextError::ChannelError(ChannelError::InvalidOrderType {
+                    expected: "Channel ordering cannot be None".to_string(),
+                    actual: chan_end_on_a.ordering.to_string(),
+                }))
             }
-            let seq_recv_path_on_b =
-                SeqRecvPath::new(&msg.packet.port_id_on_b, &msg.packet.chan_id_on_b);
-
-            client_state_of_b_on_a.verify_membership(
-                conn_end_on_a.counterparty().prefix(),
-                &msg.proof_unreceived_on_b,
-                consensus_state_of_b_on_a.root(),
-                Path::SeqRecv(seq_recv_path_on_b),
-                msg.packet.seq_on_a.to_vec(),
-            )
-        } else {
-            let receipt_path_on_b = ReceiptPath::new(
-                &msg.packet.port_id_on_b,
-                &msg.packet.chan_id_on_b,
-                msg.packet.seq_on_a,
-            );
-
-            client_state_of_b_on_a.verify_non_membership(
-                conn_end_on_a.counterparty().prefix(),
-                &msg.proof_unreceived_on_b,
-                consensus_state_of_b_on_a.root(),
-                Path::Receipt(receipt_path_on_b),
-            )
         };
```

Comment:
```suggestion
        };

```

Just to stay consistent to your change at `timeout_on_close.rs`.